### PR TITLE
download dataset/collection as bagit

### PR DESCRIPTION
**Pull request by robkooper**
there is now an optional parameter to download dataset/collection as a bagit zip file.

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [X] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [X] I have updated the CHANGELOG.md.
- [ ] I have signed the CLA
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by robkooper** [comment]
To test this:
1. create a dataset
2. add one or more files/folders to the dataset
3. create collection
4. add one ore more datasets + collections to collection

download using curl:

```
curl -o bagit.zip http://localhost:9000/api/collections/YOURID/download\?bagit\=false\&compression\=1\&key\=r1ek3rs
```

output looks approximately like:

```
Archive:  bagit.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
  3729064  01-30-2022 22:11   test_col/test_ds/folder1//DJI_0034.JPG
  3977058  01-30-2022 22:11   test_col/test_ds/folder1//DJI_0035.JPG
---------                     -------
  7706122                     2 files
```

change the bagit=true to download as bagit file and output looks like:

```
Archive:  bagit.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
      334  01-30-2022 22:22   test_col/test_col_info.json
       70  01-30-2022 22:22   test_col/test_col_metadata.json
      285  01-30-2022 22:22   test_col/test_ds/test_ds_info.json
        3  01-30-2022 22:22   test_col/test_ds/test_ds_metadata.json
      307  01-30-2022 22:22   test_col/test_ds/folder1//DJI_0034.JPG_info.json
        3  01-30-2022 22:22   test_col/test_ds/folder1//DJI_0034.JPG_metadata.json
  3729064  01-30-2022 22:22   test_col/test_ds/folder1//DJI_0034.JPG
      307  01-30-2022 22:22   test_col/test_ds/folder1//DJI_0035.JPG_info.json
        3  01-30-2022 22:22   test_col/test_ds/folder1//DJI_0035.JPG_metadata.json
  3977058  01-30-2022 22:22   test_col/test_ds/folder1//DJI_0035.JPG
       55  01-30-2022 22:22   test_col/bag-info.txt
      250  01-30-2022 22:22   test_col/bagit.txt
      524  01-30-2022 22:22   test_col/manifest-md5.txt
      139  01-30-2022 22:22   test_col/tagmanifest-md5.txt
---------                     -------
  7708402                     14 files

File: app/api/Datasets.scala [hunk 1]
```diff
@@ -2541,6 +2535,105 @@ class  Datasets @Inject()(
...
+    val byteArrayOutputStream = new ByteArrayOutputStream(chunkSize)
+    val zip = new ZipOutputStream(byteArrayOutputStream)
+    zip.setLevel(compression)
+
+    var file_index = 0
+    val buffer = new Array[Byte](chunkSize)
+    var is: Option[InputStream] = None
+
+    Enumerator.generateM({
+      println("getting chunk")
```
Comment (lmarini) [inline app/api/Datasets.scala:2586]: Should we replace all the `println` with `Logger.debug` or `Logger.trace`?

**Review by lmarini — CHANGES_REQUESTED** [review]
This is awesome. I left a minor comment regarding `println`. Also pushed a fix for the download button on the dataset page. It would throw an error since it was missing the bagit parameter and tried to parse the next as an int. I set the default in the UI to no bagit since most people find it confusing. Let me know if you think it best the other way around.

File: app/api/Datasets.scala (hunk 1 above)
Comment (robkooper) [reply app/api/Datasets.scala:2586]: this was supposed to be deleted, guess I missed a few.

**Comment by tcnichol** [comment]
I have tested this and it works as described. Marking this approved.

**Review by lmarini — APPROVED** [review]

**Pull request merged**